pallets/phala/src/compute/vault.rs

```diff
@@ -0,0 +1,530 @@
+pub use self::pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	#[cfg(not(feature = "std"))]
+	use alloc::format;
+	#[cfg(feature = "std")]
+	use std::format;
+
+	use crate::balance_convert::{div as bdiv, mul as bmul, FixedPointConvert};
+	use crate::base_pool;
+	use crate::computation;
+	use crate::pool_proxy::{ensure_stake_pool, ensure_vault, PoolProxy, PoolType, Vault};
+	use crate::registry;
+	use crate::stake_pool_v2;
+	use crate::wrapped_balances;
+
+	use crate::BalanceOf;
+	use frame_support::{
+		dispatch::DispatchResult,
+		pallet_prelude::*,
+		traits::{tokens::nonfungibles::InspectEnumerable, StorageVersion, UnixTime},
+	};
+	use frame_system::{pallet_prelude::*, Origin};
+
+	use sp_runtime::{traits::Zero, Permill, SaturatedConversion};
+	use sp_std::{collections::vec_deque::VecDeque, fmt::Display, prelude::*, vec};
+
+	pub use rmrk_traits::primitives::{CollectionId, NftId};
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config
+		+ crate::PhalaConfig
+		+ registry::Config
+		+ computation::Config
+		+ pallet_rmrk_core::Config
+		+ base_pool::Config
+		+ pallet_assets::Config
+		+ pallet_democracy::Config
+		+ wrapped_balances::Config
+		+ stake_pool_v2::Config
+	{
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		#[pallet::constant]
+		type InitialPriceCheckPoint: Get<BalanceOf<Self>>;
+	}
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(7);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	/// Mapping from the vault pid to its owner authority locking status
```

Comment:
Please mention when it's used. "owner authority locking status" is not very clear.

Reply:
done